sql/sql_table.cc

```diff
@@ -9139,17 +9138,27 @@ fk_check_column_changes(THD *thd, Alter_info *alter_info,
         return FK_COLUMN_RENAMED;
       }
 
+      /*
+        Field_{num|decimal}::is_equal evaluates to IS_EQUAL_NO where
+        the new_field adds an AUTO_INCREMENT flag on a column due to a
+	limitation in MyISAM/ARIA. For the purposes of FK determination
+        it doesn't matter if AUTO_INCREMENT is there or not.
+      */
+      const auto flags= new_field->flags;
```

Comment:
He had an agreeement with serg not to use `auto` in simple cases , like integers, for example.

I know it's not in coding style yet, but I would like to add it (as well other c++-related style elements that we're used to follow. We have none covered in current doc).